cmd/operator-sdk/migrate/cmd.go

```diff
@@ -67,12 +75,30 @@ func migrateRun(cmd *cobra.Command, args []string) error {
 	return fmt.Errorf("operator of type %s cannot be migrated", opType)
 }
 
+func verifyFlags() error {
+	// dep assumes the project's path under $GOPATH/src is the project's
+	// repo path.
+	if repo != "" && depManager == string(projutil.DepManagerDep) {
+		return fmt.Errorf(`--repo flag cannot be used with --dep-manger=dep`)
+	}
+
+	inGopathSrc, err := projutil.WdInGoPathSrc()
+	if err != nil {
+		return err
+	}
+	if !inGopathSrc && repo == "" && depManager == string(projutil.DepManagerGoMod) {
+		return fmt.Errorf(`depedency manger "modules" requires --repo be set if wd not in $GOPATH/src`)
+	}
```

Comment:
~Do we need to check the case of `!inGopathSrc && depManager == string(projutil.DepManagerDep)`?~

Nevermind, I see that we check for this in the `checkDepManagerForCmd`. I find it a little confusing that these checks are in different places. 

Would it make sense to move this logic to the `"new", "migrate"` case in `checkDepManagerForCmd`? That would have the added benefit of not duplicating this code in the `migrate` command package.

Reply:
I have this here because it's a flag check. `checkDepManagerForCmd` is supposed to be relevant for all commands, while only `new` and `migrate` have `--repo`.

Reply:
Reconsidering this position: https://github.com/operator-framework/operator-sdk/pull/1475#discussion_r293923739